pkg/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/pulumi/pulumi/pkg/v3
 
-go 1.20
+go 1.21.0
```

Comment:
It's not great to bump to 1.21 here, providers still have build time dependencies on Go 1.20 and our official support position has been well work with 1.20 and 1.21. If this is hard to revert we'll just have to bite it and hope the fallouts ok.